### PR TITLE
fix(curriculum) : update head step 4

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

As mentioned in https://github.com/freeCodeCamp/freeCodeCamp/pull/55871#issuecomment-2293607745, the tests needed updating so I updated them.

### CODE REVIEW

**Comment by lasjorg** [comment]
Thanks for updating this.

Full disclosure, the reason why I never updated this challenge is because the regex I was using might have some issues (might be catastrophic backtracking). If you remove the closing `</head>` tag and run the tests it can lock up.

I was going to update the tests with something better but never got around to it. I'm still not sure what the best solution is but it might be that we have to match on the specific elements and not `(?:.|\s*)*?`


Example of code that causes issues with the regex
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>City Skyline</title>
    <link rel="stylesheet" href="styles.css">
  <body>
  </body>
</html>
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-variables-by-building-a-city-skyline/5d822fd413a79914d39e98cc.md [hunk 1]
```diff
@@ -14,13 +14,20 @@ Within the `head`, nest a `meta` element with a `charset` of `UTF-8`, a `title`
 You should create a `meta` element within the `head` element.
 
 ```js
-assert.exists(document.querySelector('head > meta'));
+const headContentRegex = /(?<=<head\s*>)(?:.|\s*)*?(?=<\/head\s*>)/;
```
Comment (lasjorg) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-variables-by-building-a-city-skyline/5d822fd413a79914d39e98cc.md:17]:
```suggestion
const headContentRegex = /(?<=<head\s*>)[\S|\s]*(?=<\/head\s*>)/;
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-variables-by-building-a-city-skyline/5d822fd413a79914d39e98cc.md [hunk 2]
```diff
@@ -33,7 +40,12 @@ assert.exists(title);
 The `title` element should be within the `head` element.
 
 ```js
-assert.exists(document.querySelector('head > title'));
+const headContentRegex = /(?<=<head\s*>)(?:.|\s*)*?(?=<\/head\s*>)/;
```
Comment (lasjorg) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-variables-by-building-a-city-skyline/5d822fd413a79914d39e98cc.md:43]:
```suggestion
const headContentRegex = /(?<=<head\s*>)[\S|\s]*(?=<\/head\s*>)/;
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-variables-by-building-a-city-skyline/5d822fd413a79914d39e98cc.md [hunk 3]
```diff
@@ -53,41 +65,46 @@ assert.equal(title.text, 'City Skyline');
...
 ```js
-assert(document.querySelectorAll('link').length === 1);
+assert.strictEqual(document.querySelectorAll('link')?.length, 1);
 ```
 
 Your `link` element should be within your `head` element.
 
 ```js
-assert.exists(document.querySelector('head > link'));
+const headContentRegex = /(?<=<head\s*>)(?:.|\s*)*?(?=<\/head\s*>)/;
```
Comment (lasjorg) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-variables-by-building-a-city-skyline/5d822fd413a79914d39e98cc.md:87]:
```suggestion
const headContentRegex = /(?<=<head\s*>)[\S|\s]*(?=<\/head\s*>)/;
```

**Review by lasjorg — CHANGES_REQUESTED** [review]
This should work and avoid the issue the old regex had.

**Review by lasjorg — APPROVED** [review]
LGTM

**Review by Sembauke — APPROVED** [review]

**Pull request merged**